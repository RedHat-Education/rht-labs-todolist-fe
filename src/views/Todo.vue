<template>
  <div>
    <md-card>

      <md-card-header class="remove-padding">
        <Header/>
      </md-card-header>

      <md-card-content>
        TODOS WILL GO HERE.
        Probs...
          <NewTodo placeholderMsg="Add a new todos here"/>
          <ListOfTodos todos="this.todos"/>
      </md-card-content>

      <div class="image-footer"> 
        <img class="image-footer-logo" src="../assets/oil-protobadge.png" alt="Open Innovation Labs logo" />
      </div>

    </md-card>
  </div>
</template>

<script>
// @ is an alias to /src
import Header from "@/components/Header.vue";
import NewTodo from "@/components/NewTodo.vue";
import ListOfTodos from "@/components/ListOfTodos.vue";
import EventBus from "@/services/EventBus";
import { mapMutations } from 'vuex'

export default {
  name: "Todo",
  components: {
    Header,
    ListOfTodos,
    NewTodo
  },
  methods: {
    ...mapMutations([
      'editTodo',
      'toggleTodo',
      'deleteTodo'
    ]),
  },
  // data () {
  //   return { todos: []}
  // },
  // computed: {
  //   todos () {
  //     return this.$store.getters.todos
  //   }
  // }
};
</script>

<style lang="scss" scoped>
.remove-padding {
  padding: 0px;
}
.md-card {
  width: 480px;
  margin: 4px;
  display: inline-block;
  vertical-align: top;
}
.image-footer {
  border-top: solid;
  margin-top: 1px;
}
.image-footer-logo {
  padding: 5px 0px 5px 0px;
  height: 55px;
}
</style>
