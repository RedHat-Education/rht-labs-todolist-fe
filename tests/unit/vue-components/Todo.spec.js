import { shallow } from "@vue/test-utils";
import Todo from "@/views/Todo.vue";
import Header from "@/components/Header.vue";
import XofYItems from "@/components/XofYItems.vue";
import * as all from "../setup.js";

describe("Todo.vue", () => {
  it("should render like the snapshot", () => {
    const wrapper = shallow(Todo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("render with content such as header", () => {
    const wrapper = shallow(Todo);
    expect(wrapper.find(Header)).toBeTruthy();
  });
  it("render with content such as XofYItems", () => {
    const wrapper = shallow(Todo);
    expect(wrapper.find(XofYItems)).toBeTruthy();
  });
});
