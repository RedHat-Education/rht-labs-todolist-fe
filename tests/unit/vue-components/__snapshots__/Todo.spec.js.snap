// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo.vue should render like the snapshot 1`] = `
<div>
  <!---->
</div>
`;
